samples/pubsub.py

```diff
@@ -18,6 +18,7 @@
 parser = argparse.ArgumentParser(description="Send and receive messages through and MQTT connection.")
 parser.add_argument('--endpoint', required=True, help="Your AWS IoT custom endpoint, not including a port. " +
                                                       "Ex: \"abcd123456wxyz-ats.iot.us-east-1.amazonaws.com\"")
+parser.add_argument('--port', help="The port to connect to when using MQTT/TLS. AWS IoT supports 443 and 8883", type=int)
```

Comment:
Just curious, what's the use case for passing a custom port?

The samples should only cover the common supported use cases for connecting to IoT Core. The sample is already pretty complex and tough for customers to follow. I wouldn't want to add anything that isn't absolutely necessary

Reply:
As I explained in the issue, the current code will always use port 443 if the OS supports ALPN. While this work in most cases, it does not if you try to use this client to connect to Greengrass MQTT broker, since it only uses port 8883.
https://github.com/aws/aws-iot-device-sdk-python-v2/blob/7bd47b56226352ce815d82c8f8f0fd3ab8006fae/awsiot/mqtt_connection_builder.py#L149

Reply:
got it. Thanks for explaining